src/__test__/utils/plotSpecs/generateViolinSpec.test.js

```diff
@@ -0,0 +1,161 @@
+import _ from 'lodash';
+
+import { generateData } from '../../../utils/plotSpecs/generateViolinSpec';
+
+const cellSets = {
+  hierarchy: [
+    {
+      key: 'louvain',
+      children: [{ key: 'cluster-a' }, { key: 'cluster-b' }, { key: 'cluster-c' }],
+      cellIds: new Set(),
+    },
+    {
+      key: 'sample',
+      children: [{ key: 'sample-1' }, { key: 'sample-2' }],
+      cellIds: new Set(),
+    },
+    {
+      key: 'scratchpad',
+      children: [{ key: 'scratchpad-a' }],
```

Comment:
I notice these get copy-pasted a lot -- should we extract them in a single file/place and reuse them from there for each of the components?

Reply:
That's some discussion I had with myself :-)
On one hand, I hate code duplication. On the other, it is easier to refer/verify the expected values if you have them in the same file. But the same could be said about snapshots and we are happy to use them and avoid having the cluttering our test...

So, yes, I am going to (partially) extract this into a separate file.